### PR TITLE
Add edit note dialog to the program admin application page

### PR DESCRIPTION
**Pull request by clouser2**
### Description

This is present when application statuses are available. Presently, it's wired to a controller that immediately returns success (with a TODO to implement).

While here, also fixed an issue with `FieldWithLabel` where `setRows` / `setCols` wasn't actually setting the attributes correctly.

### Testing notes
Temporarily enabled status tracking and let CI run on it, which succeeded:
https://github.com/civiform/civiform/runs/7926402730?check_suite_focus=true

### Screenshots

Button in the UI:
Screen Shot 2022-08-19 at 1 17 32 PM

Modal shown:
Screen Shot 2022-08-19 at 1 17 38 PM

## Release notes:

Adding "Edit note" dialog to the program admin application page when application status tracking is enabled.

### Checklist

- [X] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)

#2912

### CODE REVIEW

**Review by rahaghas — APPROVED** [review]

File: browser-test/src/admin_application_statuses.test.ts [hunk 1]
```diff
@@ -54,14 +54,15 @@ describe.skip('view program statuses', () => {
...
 
-    it('does not Show status options', async () => {
+    it('does not show status options or edit note', async () => {
       await loginAsProgramAdmin(pageObject)
 
       await adminPrograms.viewApplications(programWithoutStatusesName)
 
       await adminPrograms.viewApplicationForApplicant(userDisplayName())
 
       expect(await adminPrograms.isStatusSelectorVisible()).toBe(false)
```
Comment (shanemc-goog) [inline browser-test/src/admin_application_statuses.test.ts:64]: related to the discussion we had on the previous PR, I think the setup here should be in the `describe` `beforeAll`, and then each expect in its own `it`.  Then they can fail individually and we get clear feedback around which is failing, and without this selector failure shadowing if the Edit Note is working or not.

File: server/conf/routes [hunk 2]
```diff
@@ -95,6 +95,7 @@ GET     /admin/programs/:programId/applications/all                       contro
 GET     /admin/programs/:programId/applications/singleVersion             controllers.admin.AdminApplicationController.downloadSingleVersion(request: Request, programId: Long)
 GET     /admin/programs/:programId/applications/:applicationId            controllers.admin.AdminApplicationController.show(request: Request, programId: Long, applicationId: Long)
 POST     /admin/programs/:programId/applications/:applicationId/updateStatus            controllers.admin.AdminApplicationController.updateStatus(request: Request, programId: Long, applicationId: Long)
+POST     /admin/programs/:programId/applications/:applicationId/editNote            controllers.admin.AdminApplicationController.editNote(request: Request, programId: Long, applicationId: Long)
```
Comment (shanemc-goog) [inline server/conf/routes:98]: ? should we use update like with updateStatus?  edit feels better in english but I wonder if we are meant to adhere to a smaller set of RESTful verbs.

Comment (clouser2) [reply server/conf/routes]: Converted to updateNote for parity. IMO, paths aren't what give RESTfulness, it's the associated HTTP verb. A truly "RESTful" setup here would be a `POST` to `/admin/programs/:programId/applications/:applicationId/notes`. We don't end up following that convention elsewhere, so likely not worth converting to it here.

Comment (shanemc-goog) [reply server/conf/routes]: Ah true true.  I think I was miss using RESTful to get at having common/readable terms in our public controller methods.

**Review by shanemc-goog — APPROVED** [review]

**Comment by shanemc-goog** [comment]
FYI I merged in the update that turns the browser tests back on. You likely won't notice a diff even if there is one as the default is still true, but I have a issue to fix that.

**Comment by clouser2** [comment]
> FYI I merged in the update that turns the browser tests back on. You likely won't notice a diff even if there is one as the default is still true, but I have a issue to fix that.

Thanks for the info. Did a fresh sync and will merge if tests continue to pass.

**Pull request merged**